### ISSUE TITLE
Interference between swifter and logging packages

### ISSUE THREAD

**Issue by raphaelvannson** [jmcarpenter2/swifter#130]
Hello,

Thank you so much for creating swifter. It is the package many of us were waiting to run a multithreaded `apply()` on a dataframe.

I am using the swifter to vectorize a function rather than creating / updating a new column of the dataframe. In this context, you can consider:
 
 1. The columns contain the function arguments.
 2. Each "instance" of the function does something in isolation (ex: process the arguments and write the result to a pickle file with its own name so that the threads do not concurrently access the same file).

I am running into an odd behavior when the `logging` module is initialized and the used by the function. It appears some processing is duplicated and -in some cases-the values of the arguments fetched by the function appear incorrect.

I am using these versions (I reverted to pandas 0.24.2, I was having trouble making swifter work with the latest v1.x.y.z).

```
Python 3.7.2
swifter             1.0.3
pandas              0.24.2
```

Here is a simple example I ran in a Jupyter notebook as I was debugging:

```python
from tabulate import tabulate
import logging
from importlib import reload
import swifter

# Initialize logging
reload(logging)
logging.basicConfig(
    level = 'DEBUG',
    format = '[%(asctime)s] [%(levelname)s]: %(message)s',    
    datefmt = '%Y-%m-%d %H:%M:%S %z') 
log_output = 'stderr'


# Define a small df
df = pd.DataFrame({'a': [1, 2, 3, 4], 'b': [11, 12, 13, 14]})
logging.info('\n%s', tabulate(df, headers='keys', tablefmt='psql'))
```

```
[2020-09-29 11:25:24 -0700] [INFO]: 
+----+-----+-----+
|    |   a |   b |
|----+-----+-----|
|  0 |   1 |  11 |
|  1 |   2 |  12 |
|  2 |   3 |  13 |
|  3 |   4 |  14 |
+----+-----+-----+
```

```python
# Define a function
def foo(x, y):
    logging.info('%s, %s', x, y)

# Apply the function to the df
foo = df.swifter.apply(lambda row: foo(row['a'], row['b']), axis = 1)
```

Output:

```
[2020-09-29 11:25:49 -0700] [INFO]: 0    1
1    2
2    3
3    4
Name: a, dtype: int64, 0    11
1    12
2    13
3    14
Name: b, dtype: int64
[2020-09-29 11:25:49 -0700] [INFO]: 1, 11
[2020-09-29 11:25:49 -0700] [INFO]: 2, 12
[2020-09-29 11:25:49 -0700] [INFO]: 3, 13
[2020-09-29 11:25:49 -0700] [INFO]: 4, 14
[2020-09-29 11:25:49 -0700] [INFO]: 1, 11
[2020-09-29 11:25:49 -0700] [INFO]: 1, 11
[2020-09-29 11:25:49 -0700] [INFO]: 1, 11
Pandas Apply: 100%
4/4 [00:00<00:00, 153.54it/s]
[2020-09-29 11:25:49 -0700] [INFO]: 1, 11
[2020-09-29 11:25:49 -0700] [INFO]: 2, 12
[2020-09-29 11:25:49 -0700] [INFO]: 3, 13
[2020-09-29 11:25:49 -0700] [INFO]: 4, 14
```

I would have expected to see only:

```
[2020-09-29 11:25:49 -0700] [INFO]: 1, 11
[2020-09-29 11:25:49 -0700] [INFO]: 2, 12
[2020-09-29 11:25:49 -0700] [INFO]: 3, 13
[2020-09-29 11:25:49 -0700] [INFO]: 4, 14
```

Is there something I am doing wrong or is this an issue?

Thanks again for this great package!

**Comment by jmcarpenter2** [comment]
Hey @raphaelvannson,

Thanks for raising this issue!

The swifter algorithm runs test pandas applies on subsets of data to estimate pandas or Dask/Modin will be faster on the full dataframe. In the algorithm, there is some code for suppressing stderr and stdout during those test applies, but the code doesn't do anything about logging. This is a really good point though -- logging should also be suppressed during the test applies so that this sort of issue doesn't arise. I'll see if I can add this to the stderr/stdout suppression code and I'll follow up here if/when a new version is released.

Thanks again Raphael!
Jason

**Comment by raphslalom** [comment]
Aaaah!! That makes perfect sense now!
Thanks for the update @jmcarpenter2 !
Glad this is useful!

**Comment by jmcarpenter2** [comment]
@raphaelvannson,

I just released swifter version 1.0.7, which solves this issue, among some other minor tweaks.

To upgrade, please `pip install swifter==1.0.7`.

Thanks again for raising this issue!
Jason

**Comment by raphslalom** [comment]
Super quick! Thanks!
